modules/opentelemetry/src/main/scala/zio/opentelemetry/tracing/SpanUtils.scala

```diff
@@ -0,0 +1,22 @@
+package zio.opentelemetry.tracing
+
+import io.opentelemetry.trace._
+import zio.clock.Clock
+import zio.{ Cause, UIO, URIO }
+
+private[opentelemetry] object SpanUtils {
+
+  /**
+   * Sets the status of `span` to `UNKNOWN` error with description being the pretty-printed cause.
+   */
+  def setErrorStatus(span: Span, cause: Cause[_]): UIO[Unit] =
```

Comment:
How about adding the capability to set a custom error status?

Reply:
That's a good suggestion

Reply:
Maybe use a partial function which maps the error `E` into a `Status`, so users can decide themselves which particular kind of their error `E` belongs to which status?

For example
```scala

def spanFrom[C, R, E, A](
    httpTextFormat: HttpTextFormat,
    carrier: C,
    getter: HttpTextFormat.Getter[C],
    spanName: String,
    spanKind: Span.Kind = Span.Kind.INTERNAL,
    toErrorStatus: PartialFunction[E, Status] = Map.empty
  )

private def setErrorStatus[E](span: Span, cause: Cause[E], toErrorStatus: PartialFunction[E, Status]): UIO[Unit] = {
    val errorStatus: Status = cause.failureOption.flatMap(toErrorStatus.lift).getOrElse(Status.UNKNOWN)
    UIO(span.setStatus(errorStatus.withDescription(cause.prettyPrint)))
  }
```

Reply:
That's a good idea!

Reply:
Fixed with https://github.com/zio/zio-telemetry/pull/119/commits/8fdf9fbc403aba42e656abda526196700dc99fb7